# 班级人员分布


> - 本模块主要使用清洗完毕的数据，对其进去再次处理，提取出班级与姓名特征，并进行去重，排序去重
> - 完毕后使用pyecharts进行折线图与饼图的绘制，最后基于班级人员分布对下步工作进行分析

In [1]:
from pyecharts import Pie
from pyecharts import Bar
import pandas as pd
import numpy as np

### 将提取的有用数据进行去重，清洗

In [2]:
data=pd.read_excel('../File/output/清洗完毕.xlsx')
data=data[['班级','姓名']]
word=pd.DataFrame(data[['班级','姓名']])
count=word.groupby(['班级','姓名']).apply(len)
count.sort_values(ascending=False, inplace=True)
wordcount=count.reset_index(name="count")
calss=wordcount[['班级','姓名']]
calss

班级    姓名
0       数媒2201   王鹏宇
1       数媒2202    周漠
2      大数据2202   赵伟博
3       数媒2203   张晓芳
4      大数据2202   介丹阳
..         ...   ...
169  云计算中高2202   任国鹏
170     数媒2203  0高铭轩
171     数媒2203   于倩倩
172  云计算中高2201   秦邵斌
173    云计算2001    康晋

[174 rows x 2 columns]

## 根据count属性进行班级的排列

In [3]:
count=calss.groupby(['班级']).apply(len)
count.sort_values(ascending=False, inplace=True)
wordcount=count.reset_index(name="count")
wordcount

班级  count
0      数媒2201     28
1      数媒2202     25
2     大数据2202     20
3    人工智能2201     19
4     云计算2201     19
5   云计算中高2201     16
6     大数据2201      9
7   云计算中高2202      6
8      数媒2203      6
9      软件2203      5
10     软件2204      4
11     软件2201      3
12   软件中高2201      3
13   软件中高2202      3
14   数字媒体2201      2
15    数媒 2201      1
16  大数据中高2201      1
17  软件中高 2201      1
18  软件中高 2202      1
19  人工智能2201       1
20    云计算2001      1

## 人员分布可视化图表

In [4]:
bar=Bar("班级人员分布","社團各班人员分布情況",title_color ="#c1392b")
bar.add("人數",wordcount['班级'],wordcount['count'],xaxis_label_textsize=8)
bar.render("../File/圖表/班级人员分布柱图.html")
bar

In [5]:
attr = wordcount['班级']
v1 = wordcount['count']
pie = Pie("班级人员分布",title_pos='left')
pie.add(
    "",
    attr,
    v1,
    is_label_show=True,
    is_more_utils=True,
    legend_pos="right",
    legend_orient="vertical",
    center=["35%", "55%"]
)
pie.render(path="../File/圖表/班级人员分布饼图.html")
pie

In [6]:
attr = wordcount['班级']
v1 = wordcount['count']
pie = Pie("班级人员分布", title_pos='center')
pie.add(
    "",
    attr,
    v1,
    radius=[40, 75],
    label_text_color=None,
    is_label_show=True,
    is_more_utils=True,
    legend_orient="vertical",
    legend_pos="left",
    center=["65%", "55%"]
)
pie.render(path="../File/圖表/班级人员分布饼图2.html")
pie

## 基于班级人员分布分析

### 从代码分析得出：
> 1. #### 数媒2201、2202以及大数据2202、人工智能2201的同学对IT社有更大的兴趣占饼状图的百分之五十一
> 2. #### 其次就是云计算2201和中高班的同学对IT社的兴趣仅次于上面各班占百分之二十
> 3. #### 最后其余班级的同学对IT社的兴趣可能想简单的了解一下里面的内容占百分之三十


 - 从这些数据上看得出来对于这些对IT社的感兴趣的同学
 - 我们要把握住他们对网络的喜欢，让他们继续坚持下去他们对网络技术的爱好
 - 同时带动那些想要简单了解网络技术的同学们一起努力学习